apps/shell/app.js

```diff
@@ -23,6 +25,161 @@ if (fs.existsSync('.env')) {
   dotenv.config({path: '.env'});
 }
 
+//Start Terminal Color Scheme Implementation
+//declared constants for directory paths
+const xdg_config_dir = (process.env["XDG_CONFIG_HOME"] || path.join(os.homedir(), ".config"));
+const ood_app_config_root = (process.env["OOD_APP_CONFIG_ROOT"] || '/etc/ood/config/ondemand/apps/shell');
+const userDir = path.join(xdg_config_dir, "ondemand", "apps", "shell", "themes");
+const systemDir = path.join(ood_app_config_root, "themes");
+
+//declared global variables for color schemes
+var schemeObjects;
+
+checkDirSync(userDir);
+checkDirSync(systemDir);
+
+if (process.env["SYSTEM_SCHEMES"] === true) {
+    var user = getSchemeObjects(userDir);
+    var system = getSchemeObjects(systemDir);
+
+   schemeObjects = {...getSchemeObjects(userDir), ...getSchemeObjects(systemDir)};
+
+} else {
+   schemeObjects = {...getSchemeObjects(userDir)};
+}
```

Comment:
If `getSchemeObjects` were modified to handle exceptions when using `readdirSync` and to return an empty `{}`, then you could get rid of `checkDirSync(userDir); checkDirSync(systemDir);` and this block and just do:

    var schemeObjects = {...getSchemeObjects(userDir), ...getSchemeObjects(systemDir)};